### DISCUSSION TITLE
Rationale behind `:` for labeled arguments

### DISCUSSION THREAD

**Discussion by wilkomanger** [gleam-lang/gleam#2256] [Ideas & suggestions]
Hi, I was wondering what the rationale behind using `:` for labeled arguments is, such as:
```gleam
let x = some_function(label_a: 30, label_b: 40)
```
From what I can tell, Gleam uses `:` as a type bound 'operator' (except for function return types), which doesn't really match the use of the argument labels.

It would seem more consistent to use `=`, as with `let` bindings, since in the end you're also assigning a value to a name with labeled arguments. It would look like this:
```gleam
let x = some_function(label_a = 30, label_b = 40)
```

I'm curious what the thought process behind this was!

**Comment by lpil** [comment]
It's a mix of being similar to mainstream languages that use `:`, and to avoid confusion with related languages like Erlang or Elixir where this

```gleam
let x = some_function(label_a = 30, label_b = 40)
```
is semantically the same as this
```gleam
let label_a = 30
let label_b = 40
let x = some_function(label_a = 30, label_b = 40)
```

Overall there's not much in it. Perhaps if we had thought `:` as an operator for annotating a type we wouldn't have used it for labels also.

**Reply by wilkomanger** [reply]
I see :+1: Too late to change now I suppose? :smile:

**Reply by lpil** [reply]
Yup, way too late!
